tag/map.go

```diff
@@ -126,13 +174,35 @@ func Update(k Key, v string) Mutator {
 // value of the tag associated with k with v. It inserts the
 // value if k doesn't exist already. It mutates the value
 // if k already exists.
+// Deprecated. Use UpsertWithMetadata instead.
 func Upsert(k Key, v string) Mutator {
 	return &mutator{
 		fn: func(m *Map) (*Map, error) {
 			if !checkValue(v) {
 				return nil, errInvalidValue
 			}
-			m.upsert(k, v)
+			m.upsert(k, v, PropagatingMetadata)
+			return m, nil
+		},
+	}
+}
+
+// UpsertWithMetadata returns a mutator that upserts the
```

Comment:
UpsertWithMetadata is a very verbose name for a core API.

Please consider short names. If not, user will never think about UpsertWithMetadata is a replacement for Upsert.